src/components/customer-resource-show.js

```diff
@@ -70,10 +72,25 @@ export default class CustomerResourceShow extends Component {
     });
   };
 
+  // when there are no fields being edited this is equal to 0; when
+  // any number of fields are being edited this is greater than 0;
+  // this number is stored on the instance so that the state will only
+  // update when this count is equal to or greater than 0
+  editCount = 0;
+
+  handleEditing = (edit) => {
```

Comment:
thinking about the naming of this method - how would you feel about `trackEditing`?  `handle` seems to imply an atomic action, like a form submission or a click event.  here we have `editCount` being mutated and persisting at this level, so it's like an ongoing thing.  similarly `onEdit` might be better named `registerEdit` or `signalEdit`...

it's a weird pattern you've got going here, passing state back up the chain.  i can't think of any other way to do it, though.  definitely gonna do some research on it - i'm sure other people have run into this type of thing before.

Reply:
WRT the weird pattern: your comment this morning is spot on. The only reason I didn't do this initially was because I didn't want to have a very large footprint for this PR. `window.prompt` is not so great, so I knew the logic around this functionality would change in the near future.

That near future is today though. I've been made aware that `stripes-form` supports blocking navigation, but they do not seem to use the method `react-router` suggests. I will be looking into their implementation and try to reproduce it in this PR (if it isn't too hacky).